mantidimaging/gui/windows/spectrum_viewer/presenter.py

```diff
@@ -375,8 +375,14 @@ def rename_roi(self, old_name: str, new_name: str) -> None:
         @param old_name: Name of the ROI to rename
         @param new_name: New name of the ROI
         """
-        self.view.spectrum_widget.rename_roi(old_name, new_name)
-        self.model.rename_roi(old_name, new_name)
+        try:
+            self.view.spectrum_widget.rename_roi(old_name, new_name)
+        except RuntimeError as err:
+            raise RuntimeError(f"Cannot remove ROI: {old_name}") from err
+        except KeyError as err:
+            raise KeyError(
+                f"Cannot rename {old_name} to {new_name}. Available: {list(self.view.spectrum_widget.rois.keys())}"
+            ) from err
```

Comment:
These exceptions don't get caught because, `spectrum_widget.rename_roi()` does not raise them. Probably best to do the check and raise the exceptions in `spectrum_widget.rename_roi()`. Then you don't need a try/except here.

Reply:
So in `spectrum_widget.rename_roi()` you have something like:
```
if old_name not in self.roi_dict.keys():
    raise RuntimeError(f"Try to remove non-existent ROI: {old_name}")
if new_name in self.roi_dict.keys():
    raise RuntimeError(f"New ROI name already exists: {new_name}")
```